code/game/objects/items/binoculars.dm

```diff
@@ -161,25 +172,64 @@
-				QDEL_NULL(coord)
-				break
-	else
-		to_chat(user, "<span class='notice'>TARGET ACQUIRED. LASER TARGETING IS ONLINE. DON'T MOVE.</span>")
-		var/obj/effect/overlay/temp/laser_target/LT = new (TU, laz_name, S)
-		laser = LT
-		playsound(src, 'sound/effects/binoctarget.ogg', 35)
-		while(laser)
-			if(!do_after(user, 50, TRUE, laser, BUSY_ICON_GENERIC))
-				QDEL_NULL(laser)
-				break
-
+	switch(mode)
+		if(0)
+			to_chat(user, "<span class='notice'>TARGET ACQUIRED. LASER TARGETING IS ONLINE. DON'T MOVE.</span>")
+			var/obj/effect/overlay/temp/laser_target/LT = new (TU, laz_name, S)
+			laser = LT
+			playsound(src, 'sound/effects/binoctarget.ogg', 35)
+			while(laser)
+				if(!do_after(user, 5 SECONDS, TRUE, laser, BUSY_ICON_GENERIC))
+					QDEL_NULL(laser)
+					break
+		if(1)
+			var/obj/effect/overlay/temp/laser_coordinate/LT = new (TU, laz_name, S)
+			coord = LT
+			to_chat(user, "<span class='notice'>SIMPLIFIED COORDINATES OF TARGET. LONGITUDE [coord.x]. LATITUDE [coord.y].</span>")
+			playsound(src, 'sound/effects/binoctarget.ogg', 35)
+			while(coord)
+				if(!do_after(user, 5 SECONDS, TRUE, coord, BUSY_ICON_GENERIC))
+					QDEL_NULL(coord)
+					break
+		if(2)
+			to_chat(user, "<span class='notice'>ACQUIRING TARGET. RAILGUN TRIANGULATING. DON'T MOVE.</span>")
+			if((GLOB.marine_main_ship?.rail_gun?.last_firing + 120 SECONDS) > world.time)
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>The Rail Gun hasn't cooled down yet!</span>")
+			else if(!targ_area)
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>No target detected!</span>")
+			else
+				to_chat(user, "<span class='notice'>TARGET ACQUIRED. RAILGUN IS FIRING. DON'T MOVE.</span>")
+				var/obj/effect/overlay/temp/laser_target/RT = new (TU, laz_name, S)
+				laser = RT
+				playsound(src, 'sound/effects/binoctarget.ogg', 35)
+				while(laser)
+					GLOB.marine_main_ship?.rail_gun?.fire_rail_gun(TU,usr)
+					if(!do_after(user, 5 SECONDS, TRUE, laser, BUSY_ICON_GENERIC))
+						QDEL_NULL(laser)
+						break
+		if(3)
+			to_chat(user, "<span class='notice'>ACQUIRING TARGET. ORBITAL CANNON TRIANGULATING. DON'T MOVE.</span>")
+			if((GLOB.marine_main_ship?.orbital_cannon?.last_orbital_firing + 500 SECONDS) > world.time)
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>Orbital bombardment not yet available!</span>")
+			else if(!targ_area)
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>No target detected!</span>")
+			else
+				to_chat(user, "<span class='notice'>TARGET ACQUIRED. ORBITAL CANNON IS READY TO FIRE.</span>")
+				var/obj/effect/overlay/temp/laser_target/OB = new (TU, laz_name, S)
+				laser = OB
+				playsound(src, 'sound/effects/binoctarget.ogg', 35)
+				to_chat(user, "<span class='notice'>RELEASE TO FIRE ORBITAL CANNON.</span>")
+				while(laser)
+					if(!do_after(user, 5 SECONDS, TRUE, laser, BUSY_ICON_GENERIC))
+						QDEL_NULL(laser)
+						break
```

Comment:
This doesn't do what you want, if you start to laser and instantly stop you will fire,
Instead you want something like this

You need to adjust the timing though, i put 15 seconds but you should adjust.
```suggestion
				if(!do_after(user, 15 SECONDS, TRUE, user, BUSY_ICON_GENERIC))
					QDEL_NULL(laser)
					return
				QDEL_NULL(laser)
```

Reply:
It is doing what I want. I want them to be able and hold the laser until the 'perfect' moment.

Just needs a better way of performing the shoot command other than being locked into firing.

Reply:
There should be a minimal amount of wait you must do though, as right now you can cheese it and instant fire.

Reply:
Yes, give me a 'start shooting' proc, i'll add it, and have it call the actual OB fire proc.